### PR TITLE
Dynamic inputs

### PR DESCRIPTION
**Pull request by magnificus**
gonna add make json and combine graphics to this concept as well (unless you protest)

### CODE REVIEW

File: src/classes/SocketClass.tsx [hunk 1]
```diff
@@ -523,7 +523,7 @@ export default class Socket extends PIXI.Container implements Tooltipable {
   }
 
   destroy(): void {
+    PPGraph.currentGraph.socketHoverOut(this);
     super.destroy();
-    this.getNode()?.resizeAndDraw();
```
Comment (fakob) [inline src/classes/SocketClass.tsx:527]: Are we now redrawing somewhere else when a socket is being deleted? 2 cases come to mind. Deleting an added trigger socket. Deleting a socket which existed on the node of the saved graph file, but does not exist on the new node.

**Review by fakob — APPROVED** [review]
Does this dynamically add and remove inputs when connecting to it so there is always one available? I think this is a great feature and it would definitely make sense to adapt this to others. Maybe also the math nodes?

File: src/classes/SocketClass.tsx (hunk 1 above)
Comment (magnificus) [reply src/classes/SocketClass.tsx:527]: Yes it dynamically adds when you drop a link on top of it and vice versa, i will check out the case of deleting an added trigger socket, trying to make sure we dont redraw all over the place

Comment (magnificus) [reply src/classes/SocketClass.tsx:527]: deleting added trigger socket seems ait to me, I think this will remove a lot of unneccessary draws

Comment (magnificus) [reply src/classes/SocketClass.tsx:527]: Adding to add node is a good idea, any other math nodes that would work you think?

**Pull request merged**